### PR TITLE
[WIP] Added a !image command that accesses the bing api

### LINKED ISSUE #124: !image
Add an !image query similar to !gif.

### PR DESCRIPTION
**Pull request by beaushinkle**
This PR is almost there, but nock is driving me crazy.

The `nock` in the test is from running `nock.recorder.rec()`, but I still always get live data.
Have any ideas?

closes #124

### CODE REVIEW

File: src/commands/useful/image.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import Promise from 'bluebird';
+import nconf from 'nconf';
+import R from 'ramda';
+
+import T from '../../translate';
+
+const request = Promise.promisify(require('request'));
+const r = require('request')
```
Comment (dustinblackman) [inline src/commands/useful/image.js]: What's this for?

File: src/commands/useful/image.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    url: `https://api.cognitive.microsoft.com/bing/v5.0/images/search?q=${suffix}&count=1&offset=0&mkt=en-us&safeSearch=Moderate`,
+    headers: {
+      'Ocp-Apim-Subscription-Key': nconf.get('BING_IMAGE_KEY') 
+    }
+  };
+
+  return request(options)
+    .then(R.prop('body'))
+    .then(body => JSON.parse(body))
+    .then(x => {console.log(x); return x;})
```
Comment (dustinblackman) [inline src/commands/useful/image.js]: Make sure to delete once done debugging.

File: src/commands/useful/image.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    method: 'GET',
+    url: `https://api.cognitive.microsoft.com/bing/v5.0/images/search?q=${suffix}&count=1&offset=0&mkt=en-us&safeSearch=Moderate`,
+    headers: {
+      'Ocp-Apim-Subscription-Key': nconf.get('BING_IMAGE_KEY') 
+    }
+  };
+
+  return request(options)
+    .then(R.prop('body'))
+    .then(body => JSON.parse(body))
```
Comment (dustinblackman) [inline src/commands/useful/image.js]: If you set `json: true` in your `options`, you can remove this.

File: src/commands/useful/image.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    .then(x => {console.log(x); return x;})
+    .then(R.prop('value'))
+    .then(R.nth(0))
+    .then(R.prop('contentUrl'))
+    .then(contentUrl => request(contentUrl))
```
Comment (dustinblackman) [inline src/commands/useful/image.js]: Replace this and the remaining of your promise chain with this, to make it a bit smaller. :)

```javascript
...
.then(R.prop('contentUrl'))
.then(request)
.then(R.path(['request', 'uri', 'href']))
```

File: tests/useful/image.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+chai.should();
+
+const original_request_fixture = fs.readFileSync(path.join(__dirname, '../fixtures/image.json'));
+
+describe.only('image', () => {
+  before(() => {
+    nconf.set('BING_IMAGE_KEY', '123');
+  });
+  nock.cleanAll()
+  nock('https://api.cognitive.microsoft.com:443', {"encodedQueryParams":true})
```
Comment (dustinblackman) [inline tests/useful/image.js]: I think your problem here is the `:443`. Removing it should fix your problem.

File: tests/useful/image.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    .get('/bing/v5.0/images/search')
+    .query({"q":"cats","count":"1","offset":"0","mkt":"en-us","safeSearch":"Moderate"})
```
Comment (dustinblackman) [inline tests/useful/image.js]: If this is giving you trouble, you could actually put the full query string in `.get()`. E.g

```
...
.get('/bing/v5.0/images/search?q=WORDS&count=1&offset=0&mkt=en-us&safeSearch=Moderate');
```

Comment (beaushinkle) [inline tests/useful/image.js]: I had that originally, and it didn't work either - same with having or not having 443

File: src/commands/useful/image.js (hunk 4 above)
Comment (beaushinkle) [inline src/commands/useful/image.js]: sweet - I need to look more into all the R helpers

File: tests/useful/image.js (hunk 6 above)
Comment (dustinblackman) [inline tests/useful/image.js]: Looking at your code, you're technically making two requests and you only have one request nocked. That could potentially be your issue depending if the host has changed compared to what you have nocked currently.

**Pull request closed without merging**